pkg/prometheus/promcfg.go

```diff
@@ -727,6 +731,16 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 	cfg = enforcer.addLimitsToYAML(cfg, labelNameLengthLimitKey, m.Spec.LabelNameLengthLimit, enforcedLabelNameLengthLimit)
 	cfg = enforcer.addLimitsToYAML(cfg, labelValueLengthLimitKey, m.Spec.LabelValueLengthLimit, enforcedLabelValueLengthLimit)
 
+	// Since BodySizeLimit is defined only in PrometheusCRD
+	if enforcedBodySizeLimit != "" {
+		if version.Major == 2 && version.Minor >= 28 {
```

Comment:
(nit) more consistent and robust (e.g. it will continue to work with v3.x if it still supports the field)
```suggestion
		if version.GTE(semver.MustParse("2.28.0")) {
```